Date.js

```diff
@@ -0,0 +1,45 @@
+function convertEpochToDateTime(epoch){
+	return new Date(epoch);
+}
+
+function convertDateTimeToEpoch(dateTime){
+	return dateTime.getTime();
+}
+
+function getSundayOfCurrentWeekInEpoch(todaysDateInDateTime){
+	var dayOfWeek = todaysDateInDateTime.getDay();
+	return new Date(todaysDateInDateTime.getTime() - dayOfWeek*86400000).setHours(0,0,0,0);
+}
+
+function getSaturdayOfCurrentWeekInEpoch(todaysDateInDateTime){
+	var dayOfWeek = todaysDateInDateTime.getDay();
+	return new Date(todaysDateInDateTime.getTime() + (6-dayOfWeek)*86400000).setHours(23,59,59,59);
+}
+
+function getSundayOfPreviousWeekInEpoch(currentlyDisplayedSundayInEpoch){
+	return new Date(currentlyDisplayedSundayInEpoch - 7*86400000).setHours(0,0,0,0);
+}
+
+function getSaturdayOfPreviousWeekInEpoch(currentlyDisplayedSundayInEpoch){
+	return new Date(currentlyDisplayedSundayInEpoch - 1*86400000).setHours(0,0,0,0);
+}
+function getSundayOfNextWeekInEpoch(currentlyDisplayedSundayInEpoch){
+	return new Date(currentlyDisplayedSundayInEpoch + 7*86400000).setHours(0,0,0,0);
+}
+
+function getSaturdayOfNextWeekInEpoch(currentlyDisplayedSundayInEpoch){
+	return new Date(currentlyDisplayedSundayInEpoch + 13*86400000).setHours(0,0,0,0);
+}
+function getSelectedDateInCurrentWeekInEpoch(todaysDate, dayOfWeekNumber, time) {
```

Comment:
since you are dealing with `Epoch` time throughout, drop `Epoch` everywhere do a string replace in the file using your editor and just name things like: `getSundayOfNextWeek`, `getSaturdayOfNextWeek` etc